diff --git a/caliper-config.cmake.in b/caliper-config.cmake.in
index c3a6f947..cd177ef3 100644
--- a/caliper-config.cmake.in
+++ b/caliper-config.cmake.in
@@ -2,6 +2,11 @@
 
 include(CMakeFindDependencyMacro)
 
+if (@CALIPER_HAVE_ROCPROFILER@)
+  find_dependency(rocprofiler-sdk CONFIG)
+  find_dependency(rocprofiler-sdk-roctx CONFIG)
+endif()
+
 include("${CMAKE_CURRENT_LIST_DIR}/caliper-targets.cmake")
 
 set_and_check(caliper_INCLUDE_DIR "@PACKAGE_caliper_INSTALL_INCLUDE_DIR@")
